backend/src/main/java/com/daedan/festabook/place/service/PlaceService.java

```diff
@@ -77,6 +78,26 @@ public void deleteByPlaceId(Long placeId) {
         placeJpaRepository.deleteById(placeId);
     }
 
+    @Transactional
+    public void updatePlace(Long placeId, PlaceUpdateRequest request) {
+        Place place = getPlaceByPlaceId(placeId);
+        place.updateCategory(request.placeCategory());
+
+        PlaceDetail newPlaceDetail = request.toPlaceDetail(place);
+        PlaceDetail existingDetail = placeDetailJpaRepository.findByPlaceId(placeId).orElse(null);
+
+        updatePlaceDetail(existingDetail, newPlaceDetail);
+    }
```

Comment:
```suggestion
    public void updatePlace(Long placeId, PlaceUpdateRequest request) {
        Place place = getPlaceByPlaceId(placeId);
        place.updateCategory(request.placeCategory());
        
        PlaceDetail newDetail = request.toPlaceDetail(place);
        upsertPlaceDetail(placeId, newDetail);
    }
    
    private void upsertPlaceDetail(Long placeId, PlaceDetail newDetail) {
        Optional<PlaceDetail> optionalDetail = placeDetailJpaRepository.findByPlaceId(placeId);

        if (optionalDetail.isPresent()) {
                PlaceDetail existing = optionalDetail.get();
                existing.update(newDetail);
                return;
        }
    
        placeDetailJpaRepository.save(newDetail);
}
```
개선 로직 어떤가요? 내부에서 찾아서 처리하는 형식으로 가독성을 향상 시킬 수 있을 거 같아요!